add scala_platform to export output for metals!

cosmicexplorer opened the pull request:
### Problem

We would like to support the Metals VSCode plugin -- see associated PR at scalameta/metals#935. Pants currently does not have any fields which specifically provide the scala version and the compiler jars, so that PR currently iterates over the `libraries` keys, which doesn't provide all the scala compiler jars.

### Solution

Add a `scala_platform` key to the `./pants export` output containing the `scala_version` and `compiler_classpath`:
```json
{
  ...,
  "scala_platform": {
    "scala_version": "2.12",
    "compiler_classpath": [
      "/path/to/scala-compiler-2.12.8.jar",
      "/path/to/scala-library-2.12.8.jar",
      ...
    ]
}
```

### Result

Using the branch at https://github.com/cosmicexplorer/language-server/tree/sohamr/add-pants-build-tool (which is based off of scalameta/metals#935), I have been able to show that metals can extract the `scala_platform` from the json!

wisechengyi commented:
Thanks, @cosmicexplorer ! looks good. One thing we need to bump the export version from 
`1.0.10 to 1.0.11`:
https://github.com/pantsbuild/pants/blob/51aca069b548836435504de4b273188acf1c0a15/src/python/pants/backend/project_info/tasks/export.py#L58
https://github.com/pantsbuild/pants/blob/23a8a2b99ac8bdae39485d98f6666fc0793c1d77/tests/python/pants_test/backend/project_info/tasks/test_export.py#L252

wisechengyi commented:
Sorry one more: update the doc: https://github.com/pantsbuild/pants/blob/ecd15542eefff1d112a5ce96ac4ec3f871dbd21a/src/docs/export.md#export-format-changes

cosmicexplorer commented:
@wisechengyi bumped the export format version and added docs, and also added testing! It required adding a lot of new targets in order to allow the export task to pull the scala platform jars, which will always be injected in a normal pants build, but not in v1 task tests such as this.
